transport/conn.go

```diff
@@ -188,12 +214,18 @@ func WrapConn(conn net.Conn, s StreamIDAllocator) *Conn {
 			conn: bufio.NewReaderSize(conn, ioBufferSize),
 			h:    make(map[frame.StreamID]responseHandler),
 		},
+		cfg:    cfg,
 		stream: s,
 	}
 	go c.w.loop()
 	go c.r.loop()
 
-	return c
+	if err := c.initializeConnection(); err != nil {
+		// TODO close connection
```

Comment:
Go for it. This should be documented and Close function shall be added.

Reply:
But do we still want to initialize connection in WrapConn?